utility/serial.cpp

```diff
@@ -92,6 +92,60 @@ void ArdusatSerial::begin(unsigned long baud, bool setXbeeSpeed)
   }
 }
 
+/**
+ * Begin serial communications with a Sparkfun BlueSMiRF module at the specified baud
+ * rate.
+ *
+ * Note that baud rates above ~57600 are not well-supported by SoftwareSerial, and 
+ * even 57600 may cause some bugs.
+ *
+ * @param baud rate for serial communications
+ */
+void ArdusatSerial::beginBluetooth(unsigned long baud)
+{
+  if (_mode == SERIAL_MODE_HARDWARE || _mode == SERIAL_MODE_HARDWARE_AND_SOFTWARE) {
+    Serial.begin(baud);
+
+    _soft_serial->end();
+    _soft_serial->begin(115200); // bluesmirf defaults to baudrate of 115200
+
+    _soft_serial->print("$"); // enter command mode
+    _soft_serial->print("$");
+    _soft_serial->print("$");
+    delay(100);
+
+    _soft_serial->println("S~,0"); // Set to serial port profile
+
+    bool valid = true;
+
+    // Supported bluesmirf baud rates
+    switch(baud) { // Set Baud rate
+      case 1200: _soft_serial->println("U,1200,N");
```

Comment:
You need `break`s in here, otherwise you're writing each of these to the unit and falling to the default making `valid` always be `false`